src/app/app.component.ts

```diff
@@ -664,4 +664,20 @@ export class AppComponent implements OnInit, AfterViewInit {
   public duplicateRouteResponse() {
     this.environmentsService.duplicateRouteResponse();
   }
+
+  /**
+   * Enable/disable random response
+   */
+  public toggleRandomResponse(uuid: string) {
+    this.store.update(
+      toggleRandomResponse(uuid)
```

Comment:
please use `updateRouteAction()`
```suggestion
  public toggleRandomResponse(randomResponse: boolean) {
    this.store.update(
      updateRouteAction({ randomResponse: !randomResponse })
```